### PR TITLE
Resolve res.json() Before Setting Data

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR adds an intermediate step in the promise chain to resolve the <code>res.json()</code> promise before setting data, so that the <code>.catch()</code> is appropriately triggered if no valid JSON is found.

### CODE REVIEW

**Review by MagoDelBlocco — APPROVED** [review]
LGTM

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 1]
```diff
@@ -19,7 +19,10 @@ export const BuildSnapshotContainer = React.memo((props) =>
      * @see <a href="www.robinwieruch.de/react-hooks-fetch-data">Fetching</a>
      */
     React.useEffect(() => {
-      fetch(SOURCE).then(res => setData(res.json())).catch(setData([]));
+      fetch(SOURCE)
+      .then(res => res.json())
```
Comment (krasimir-google) [inline src/main/webapp/components/BuildSnapshotContainer.js:23]: please indent this `.` and the ones on the following lines by +2 spaces

**Review by krasimir-google — CHANGES_REQUESTED** [review]

File: src/main/webapp/components/BuildSnapshotContainer.js (hunk 1 above)
Comment (samuelsogbesan) [reply src/main/webapp/components/BuildSnapshotContainer.js]: Indented f25134f

**Review by krasimir-google — APPROVED** [review]
LGTM

Thank you!

**Pull request merged**